cmd/url.go

```diff
@@ -50,24 +53,61 @@ The created URL can be used to access the specified component from outside the O
 		checkError(err, "")
 		projectName := project.GetCurrent(client)
 
+		var app string
 		var componentName string
+		var urlName string
+
+		if urlApplication == "" {
+			app, err = application.GetCurrent(client)
+			checkError(err, "")
+		} else {
+			exists, err := application.Exists(client, urlApplication)
+			checkError(err, "unable to check if the application exists or not")
+			if !exists {
+				fmt.Printf("The application %s does not exists in the project %s\n", urlApplication, projectName)
+				os.Exit(1)
+			}
+			app = urlApplication
+		}
+
+		if len(urlComponent) == 0 {
+			componentName = getComponent(client, "", app, projectName)
```

Comment:
just call this as `getComponent(client, urlComponent, app, projectName)`
And you don't have to do any other checks (you can delete this whole `if` statement), existing check is alrady in that function